### ISSUE TITLE
MaxListenersExceededWarning in Subscription.js

### ISSUE THREAD

**Issue by PizzaPartyInc** [guidesmiths/rascal#178]
Hello!

Starting with rascal version `13.1.1` these warnings started appearing for us.

```
'(node:20716) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [Connection]. Use emitter.setMaxListeners() to increase limit\n' +
'    at _addListener (events.js:390:17)\n' +
'    at Connection.addListener (events.js:406:10)\n' +
'    at Connection.once (events.js:437:8)\n' +
'    at attachErrorHandlers (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Subscription.js:250:16)\n' +
'    at Object.callback (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Subscription.js:62:37)\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\async\\dist\\async.js:1311:26\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\async\\dist\\async.js:321:20\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\lodash\\lodash.js:10118:25\n' +
'    at callback (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Vhost.js:403:7)\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:291:27\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:59:7\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:74:5\n' +
'    at reply (E:\\PathToProject\\node_modules\\amqplib\\lib\\channel.js:106:16)\n' +
'    at ConfirmChannel.C.accept (E:\\PathToProject\\node_modules\\amqplib\\lib\\channel.js:451:12)\n' +
'    at Connection.mainAccept [as accept] (E:\\PathToProject\\node_modules\\amqplib\\lib\\connection.js:64:33)\n' +
'    at Socket.go (E:\\PathToProject\\node_modules\\amqplib\\lib\\connection.js:478:48)'

'(node:20716) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [Connection]. Use emitter.setMaxListeners() to increase limit\n' +
'    at _addListener (events.js:390:17)\n' +
'    at Connection.addListener (events.js:406:10)\n' +
'    at Connection.once (events.js:437:8)\n' +
'    at attachErrorHandlers (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Subscription.js:251:16)\n' +
'    at Object.callback (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Subscription.js:62:37)\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\async\\dist\\async.js:1311:26\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\async\\dist\\async.js:321:20\n' +
'    at E:\\PathToProject\\node_modules\\rascal\\node_modules\\lodash\\lodash.js:10118:25\n' +
'    at callback (E:\\PathToProject\\node_modules\\rascal\\lib\\amqp\\Vhost.js:403:7)\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:291:27\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:59:7\n' +
'    at E:\\PathToProject\\node_modules\\amqplib\\lib\\callback_model.js:74:5\n' +
'    at reply (E:\\PathToProject\\node_modules\\amqplib\\lib\\channel.js:106:16)\n' +
'    at ConfirmChannel.C.accept (E:\\PathToProject\\node_modules\\amqplib\\lib\\channel.js:451:12)\n' +
'    at Connection.mainAccept [as accept] (E:\\PathToProject\\node_modules\\amqplib\\lib\\connection.js:64:33)\n' +
'    at Socket.go (E:\\PathToProject\\node_modules\\amqplib\\lib\\connection.js:478:48)'
```
Looks like the fix that was done in the context of this issue https://github.com/guidesmiths/rascal/issues/164 has suppressed the warning in one case, but exposed the warnings in our...
I don't think we are doing anything out of the ordinary and having multiple subscriptions (which establish multiple channels) should be fine.

My current assumption is that before the 13.1.1 fix, `setMaxListeners` was set for `channel.connection`, while after it is set for regular `connection`, 2 of which represent different connection instances. (which is further supported by this line https://github.com/guidesmiths/rascal/blob/master/lib/amqp/Vhost.js#L405, since there is a need to assign a custom `_rascal_id` property to channel connection)

Is it possible to fix this for both cases (this and https://github.com/guidesmiths/rascal/issues/164 )? Or maybe there is a way to manually suppress these warnings without library changes?

Thanks!

**Comment by cressie176** [comment]
Hi @PizzaPartyInc,
Thanks for reporting and the thorough investigation. I think you might be right. Will see what I can do.

**Comment by cressie176** [comment]
Published as rascal v14.0.1
Please see the changelog for details of the breaking change between 13 and 14.
